File: ios/ObjectiveC/Beacon/2-Notification/Complete/NotificationComplete/MyNotificationFilter.m
//
//  MyNotificationFilter.m
//  NotificationComplete
//
//  Created by Olivier Stevens on 23/11/2017.
//  Copyright © 2017 sarra srairi. All rights reserved.
//

#import "MyNotificationFilter.h"

@implementation MyNotificationFilter

-(id) initWithMinTimeBetweenNotification:(double)minTimeBetweenNotification{
    self = [super init];
    if(self){
        self->minTimeBetweenNotification = (double) minTimeBetweenNotification;
        minNextTimeNotification = 0;
    }
    return self;
}

- (NSString * _Nonnull)getName {
    return @"MyNotificationFilter";
}

- (BOOL)createNewNotificationWithPlaceNotification:(id<PlaceNotification> _Nonnull)placeNotification {
    return minNextTimeNotification < CACurrentMediaTime();
}

- (BOOL)deleteCurrentNotificationWithPlaceNotification:(id<PlaceNotification>)placeNotification {
    return true;
}

- (void)onBackground {

}

- (void)onForeground {

}

- (void)onNotificationCreatedWithPlaceNotification:(id<PlaceNotification> _Nonnull)placeNotification created:(BOOL)created {

}

- (void)onNotificationDeletedWithPlaceNotification:(id<PlaceNotification>)placeNotification deleted:(BOOL)deleted {
    if (deleted) {
        minNextTimeNotification = CACurrentMediaTime() + minTimeBetweenNotification;
    }
}

- (void)loadWithUserDefaults:(NSUserDefaults * _Nonnull)userDefaults {
    minNextTimeNotification = [userDefaults doubleForKey:MIN_NEXT_TIME_NOTIFICATION];
}

- (void)saveWithUserDefaults:(NSUserDefaults * _Nonnull)userDefaults {
    [userDefaults setDouble:minNextTimeNotification forKey:MIN_NEXT_TIME_NOTIFICATION];
}

- (void)updateParametersWithParameters:(id _Nonnull)parameters {

}

@end

File: ios/SWIFT/Beacon/1-QuickStart/Complete/QuickStart/MyNotificationFilter.swift
//
//  MyNotificationFilter.swift
//  QuickStart
//
//  Created by Ludovic Vimont on 08/11/2017.
//  Copyright © 2017 R&D connecthings. All rights reserved.
//

import Foundation
import ConnectPlaceActions

public class MyNotificationFilter: NotificationFilter {
    let MIN_NEXT_TIME_NOTIFICATION = "com.notification.minNextTimeNotification"
    var minTimeBetweenNotification : Double!
    var minNextTimeNotification : Double!

    convenience init(timeBetweenNotification: Int) {
        self.init()
        self.minTimeBetweenNotification = (Double(timeBetweenNotification) / 1000)
    }
    
    public func getName() -> String {
        return "MyNotificationFilter"
    }
    
    public func createNewNotification(placeNotification: PlaceNotification) -> Bool {
        return self.minNextTimeNotification < CFAbsoluteTimeGetCurrent()
    }
    
    public func deleteCurrentNotification(placeNotification: PlaceNotification) -> Bool {
        return true
    }
    
    public func onNotificationCreated(placeNotification: PlaceNotification, created: Bool) {
        
    }
    
    public func onNotificationDeleted(placeNotification: PlaceNotification, deleted: Bool) {
        if deleted {
            minNextTimeNotification = CACurrentMediaTime() + minTimeBetweenNotification;
        }
    }
    
    public func onBackground() {
        
    }
    
    public func onForeground() {
        
    }
    
    func save(_ dataStore: UserDefaults) {
    }
    

    public func load(userDefaults: UserDefaults) {
        minNextTimeNotification = userDefaults.double(forKey: MIN_NEXT_TIME_NOTIFICATION)
    }
    
    public func save(userDefaults: UserDefaults) {
        userDefaults.set(minNextTimeNotification, forKey: MIN_NEXT_TIME_NOTIFICATION)
    }
    
    public func updateParameters(parameters: AnyObject) {
        
    }
}

